tests/multiple_listening_addresses.rs

```diff
@@ -30,8 +30,22 @@ fn multiple_concurrent_ephemeral_listening_addresses_on_same_ip() {
 
         let (first, second) = futures::future::join(first, second).await;
 
-        // one should succeed and other should fail
-        assert_eq!(first.is_ok(), second.is_err());
+        // before an Swarm alike api on the background task to make sure we attempt to modify the
+        // background task twice before the swarm gets to poll, this will forever produce one or two
+        // successes.
+        //
+        // with two attempts without polling the swarm in the between:
+        // assert_eq!(first.is_ok(), second.is_err());
+        //
+        // intuitively it could seem that first will always succeed because it must get the first
+        // attempt to push messages into the queue but not sure if that should be leaned on.
+
+        let successes = [first.is_ok(), second.is_ok()]
+            .iter()
+            .filter(|&&success| success)
+            .count();
```

Comment:
since we only want to know if there's more than one success, we could just write: `let successes = first.is_ok() || second.is_ok()` and `assert!(successes, ...)`

Reply:
I will change this so but I felt the counting was more obvious with the "one or two" text I had above.